Library/Homebrew/rubocops/components_order.rb

```diff
@@ -97,46 +106,49 @@ def audit_formula(_node, _class_node, _parent_class_node, body_node)
               next
             end
 
-            if on_macos_blocks.length > 1
-              problem "there can only be one `on_macos` block in a resource block."
-              next
-            end
-
-            if on_linux_blocks.length > 1
-              problem "there can only be one `on_linux` block in a resource block."
-              next
+            on_system_blocks.each do |on_method, blocks|
+              if blocks.length > 1
+                problem "there can only be one `#{on_method}` block in a resource block."
+                next
+              end
             end
           end
         end
 
-        def check_on_os_block_content(component_precedence_list, on_os_block)
-          on_os_allowed_methods = %w[
+        def check_on_system_block_content(component_precedence_list, on_system_block)
+          on_system_allowed_methods = %w[
+            url
+            version
+            sha256
+            mirror
             depends_on
             patch
             resource
             deprecate!
             disable!
             conflicts_with
+            fails_with
```

Comment:
`on_{system}` blocks can now contain `url`, `version`, `sha256`, `mirror`, and `fails_with` calls as well as other `on_{system}` blocks.

Reply:
> only raises issues with [libgccjit](https://github.com/Homebrew/homebrew-core/blob/master/Formula/libgccjit.rb) it turns out, and that formula definitely seems is a unique case.

@fxcoudert may have an insight on how long-lived this is likely to be.

> Thoughts on the best way to handle that formula?

Probably depends on @fxcoudert thoughts. I can see an argument that even if we only need it in this one case: some taps may need it in more e.g. those that ship binaries in their `url`.

Reply:
Okay, that makes sense. In the meantime, I think I'll hold off on allowing these and just temporarily not change `libgccjit`. That way, we can get a tag and start updating the rest of the formulae and casks, and then we'll just have to come back and make a decision before merging this PR (and banning `if Hardware::CPU.arm?`.

Reply:
Sounds good!